src/components/admin-management/admin-form/hooks.ts

```diff
@@ -0,0 +1,154 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import {
+  SubmitHandler,
+  UseFormGetValues,
+  UseFormSetError,
+  UseFormSetValue,
+  UseFormTrigger,
+} from 'react-hook-form';
+
+import useCopyToClipboard from 'src/hooks/useCopyToClipboard';
+import { generatePassword } from 'src/utils/generatePassword';
+
+import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query';
+import { useRouter } from 'next/router';
+import { useTranslation } from 'react-i18next';
+import { EMAIL_ERROR, PHONE_ERROR } from 'src/constants';
+import { adminPanelApi } from 'src/redux/api/adminPanelAPI';
+import { useAccountCheckMutation } from 'src/redux/api/authApi';
+import { ROUTES } from 'src/routes';
+import { AdminFormValues } from './types';
+
+interface UseAdminFormProps {
+  onSubmit: (data: AdminFormValues) => void;
+  setValue: UseFormSetValue<AdminFormValues>;
+  getValues: UseFormGetValues<AdminFormValues>;
+  setError: UseFormSetError<AdminFormValues>;
+  trigger: UseFormTrigger<AdminFormValues>;
+  isEditingExistingAdmin: boolean;
+  adminId?: string;
+  touchedFields: Partial<
+    Readonly<{
+      phoneNumber?: boolean | undefined;
+    }>
+  >;
+}
+
+interface UseAdminFormReturnType {
+  handleArrowBackClick: () => Promise<boolean>;
+  handleClickShowPassword: () => void;
+  handleGeneratePasswordClick: () => void;
+  handleCopyClick: () => Promise<boolean>;
+  copiedText: string | null;
+  resetCopiedText: () => void;
+  handleSubmitClick: SubmitHandler<AdminFormValues>;
+  showPassword: boolean;
+  isPasswordRegenerated: boolean;
+  setIsPasswordRegenerated: Dispatch<SetStateAction<boolean>>;
+}
+
+export const useAdminForm = ({
+  onSubmit,
+  setValue,
+  getValues,
+  trigger,
+  setError,
+  touchedFields,
+  isEditingExistingAdmin,
+  adminId,
+}: UseAdminFormProps): UseAdminFormReturnType => {
+  const { t: translate } = useTranslation();
+  const router = useRouter();
+  const { copiedText, copy, resetCopiedText } = useCopyToClipboard();
+
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+  const [isPasswordRegenerated, setIsPasswordRegenerated] = useState<boolean>(false);
+
+  const [accountCheck] = useAccountCheckMutation();
+
+  const [regeneratePassword] = adminPanelApi.useUpdateAdminPasswordMutation();
+
```

Comment:
two different code styles, line 67 and 69

Reply:
fixed